### DISCUSSION TITLE
Using dedicated TPU to help

### DISCUSSION THREAD

**Discussion by salleq** [nerdaxic/glados-voice-assistant#14] [Ideas]
TTS engine seems to partly at least use something that might benefit from using a dedicated TPU.

I'm no expert on this matter but the biggest delay seems to be turning the generated voice data to sound (?)

**Comment by nerdaxic** [comment]
TTS delay is the largest time suck in the process.

Was thinking the same, could be handy to add support for example [coral.ai USB Accelerator](https://coral.ai/products/accelerator).
With some luck this could allow the software to run on Raspberry Pi again. Could be as simple few lines added to [glados-tts](https://github.com/nerdaxic/glados-tts) engine.py file.

**✨ Looking for sponsors, please ship me a Coral.**

Currently I'm fine runnin mine on CPU, so I prioritise wake-word detection and home assistant integrations over this.

**Comment by salleq** [comment]
Yes, actually I forgot to mention I do have a coral.ai myself. I'm using it with Frigate (https://github.com/blakeblackshear/frigate).
I bought mine from https://www.mouser.fi/ProductDetail/Coral/G950-06809-01?qs=sGAEpiMZZMu3sxpa5v1qrv9%252Bz890pMquX2qAISzny6s%3D but delivery times are gigantic right now...like with everything else.

I understand it might not be a priority right now. So let's leave it to thinking hat for now :)
